libs/oidc/common/src/lib/services/client-metadata/client-metadata.service.ts

```diff
@@ -81,12 +92,16 @@ export class ClientMetadataService {
       redirectUri.save();
     });
 
+    clientMetadata.backchannelLogoutUris.map((backchannelUri) => {
+      backchannelUri.save();
+    });
+
```

Comment:
`backchannelUri.save()` is async and line 99 *can* execute before the save is complete. Potential race condition and the save on line 99 can fail since the relationships will yet exist.

Use `await Promise.all(promisesArray);` to ensure those relationships are inserted before proceeding. This applies to inserting client metadata above this block.